app/models/ability.rb

```diff
@@ -10,18 +8,50 @@ def initialize(user)
 
     alias_action :create, :read, :update, :destroy, to: :crud
 
+    # unconfirmed user
+    can :read, User
+    can :update, User, id: user.id
+    can :resend_confirmation_instruction, User, id: user.id
+    can :read_email, User, hide_email: false # view helper
+    can :read, Team
+    can :read, Project
+    can :read, :feed_entry
+
+    # confirmed user
     can :crud, User, id: user.id
-    can :crud, User if user.admin?
     can :resend_confirmation_instruction, User, id: user.id
-    can :resend_confirmation_instruction, User if user.admin?
+    can :read, :mailing  if signed_in?(user)
+    can :read, Mailing do |mailing|
+      mailing.recipient? user
+    end
+    can :create, Project if user.confirmed?
+
+    # current_student
+    can :crud, Conference if user.current_student?
```

Comment:
Would we need to add `&& user.confirmed?` to all the checks below to make sure they are only available to confirmed users?

I like this organization - it seems clearer to me. Mixing "confirmed" and "unconfirmed" with user roles seems a bit odd to me, though.

Update: ok, things make a bit more sense when I read the "Give permissions, don't take them away" link you posted. Doing things like `return unless user.confirmed?` right before line 20 ensures that everything below is only for confirmed users.

Reply:
Yes, @pgaspar , instead of all the `confirmed?` checks, we will have one return statement. 😍 
I agree it would be even neater to have no authentication check in here at all, but we'll probably going to need that until we went through everything in #991. 🍀 
Just a few steps away 🤣